x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -28,13 +28,20 @@ export class AddLayerPanel extends Component {
 
     this.state = {
       sourceType: null,
+      isLoading: false,
+      hasLayerSelected: false,
+    };
+  }
+
+  static getDerivedStateFromProps(props) {
+    return {
+      isLoading: props.isLoading,
+      hasLayerSelected: props.hasLayerSelected
     };
   }
 
   _previewLayer = (source) => {
-    this.layer = source.createDefaultLayer({
-      temporary: true,
-    });
+    this.layer = source.createDefaultLayer();
```

Comment:
while we're in here, let's clean this up. we shouldn't be tracking state on the component-instance. move this to the actual `this.state`.

Reply:
Ooh, that is weird. Good call